Servers/WindowServer/WSWindowFrame.cpp

```diff
@@ -227,7 +231,14 @@ static Rect frame_rect_for_window_type(WSWindowType type, const Rect& rect)
 
 Rect WSWindowFrame::rect() const
 {
-    return frame_rect_for_window_type(m_window.type(), m_window.rect());
+    auto rect = frame_rect_for_window_type(m_window.type(), m_window.rect());
+    if (!m_window.show_titlebar()) {
```

Comment:
If the window type can never have a title bar, it seems wrong to check if show_titlebar().
How about turning frame_rect_for_window_type(type, rect) into frame_rect_for_window(window) and then it could remove the titlebar rect only if it's relevant for the window's type.